### PR TITLE
13 ctra model

### PR DESCRIPTION
**Pull request by reyespinr**
# PR Details
## Description

This PR contains the implementation of the Constant Turn Rate and Acceleration model. This model describes the motion of a vehicle or object with a constant yaw rate and acceleration. The CTRA model includes a state vector `CtraState` with six state variables: x and y position, heading angle, velocity, turning rate, and acceleration. The Angle class is implemented into the motion model.

## Related Issue
Missing CTRA model: https://github.com/usdot-fhwa-stol/cooperative-perception-core/issues/13

## Motivation and Context

This motion model is used in the prediction step of the Unscented Kalman filter that will be used for temporal alignment.

## How Has This Been Tested?

Created unit test to validate the model. Results for the unit tests were validated by using Python script and manual calculations.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by MishkaMN — COMMENTED** [review]
Test

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -16,18 +16,19 @@
...
   ctrv_model.cpp
+  ctra_model.cpp
 )
 
 add_library(cooperative_perception::cooperative_perception ALIAS cooperative_perception_coreLibrary)
 
 target_include_directories(cooperative_perception_coreLibrary
   PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
```
Comment (adamlm) [inline src/CMakeLists.txt:26]: Needs an extra indent. It should look like:
```cmake
target_include_directories(cooperative_perception_coreLibrary
  PUBLIC
    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
)
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -16,18 +16,19 @@
...
 )
 
 target_link_libraries(cooperative_perception_coreLibrary
   PUBLIC
-    Boost::container
-    Eigen3::Eigen
-    units::units
+  Boost::container
+  Eigen3::Eigen
+  units::units
```
Comment (adamlm) [inline src/CMakeLists.txt:33]: Needs an extra indent. It should look like:
```cmake
target_link_libraries(cooperative_perception_coreLibrary
  PUBLIC
    Boost::container
    Eigen3::Eigen
    units::units
)
```

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -17,15 +17,16 @@
...
 )
 
 target_link_libraries(test_cooperative_perceptionExecutable
   PRIVATE
-    cooperative_perception::cooperative_perception
-    GTest::GTest
-    GTest::Main
+  cooperative_perception::cooperative_perception
+  GTest::GTest
+  GTest::Main
```
Comment (adamlm) [inline test/CMakeLists.txt:29]: Needs an extra indent. It should look like:
```cmake
target_link_libraries(test_cooperative_perceptionExecutable
  PRIVATE
    cooperative_perception::cooperative_perception
    GTest::GTest
    GTest::Main
)
```

**Review by adamlm — CHANGES_REQUESTED** [review]
Minor formatting changes. See line specific comments.

**Review by adamlm — APPROVED** [review]
Looks good to me.

**Comment by MishkaMN** [comment]
Looks good to me

**Review by MishkaMN — APPROVED** [review]

**Pull request merged**